src/documentation/programming-guide.md

```diff
@@ -1010,26 +1421,41 @@ While `ParDo` always produces a main output `PCollection` (as the return value f
 {% github_sample /apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/snippets_test.py tag:model_pardo_with_undeclared_outputs
 %}```
 
-## <a name="transforms-composite"></a>Composite Transforms
+### 4.6. Composite transforms
 
-Transforms can have a nested structure, where a complex transform performs multiple simpler transforms (such as more than one `ParDo`, `Combine`, `GroupByKey`, or even other composite transforms). These transforms are called composite transforms. Nesting multiple transforms inside a single composite transform can make your code more modular and easier to understand.
+Transforms can have a nested structure, where a complex transform performs
+multiple simpler transforms (such as more than one `ParDo`, `Combine`,
+`GroupByKey`, or even other composite transforms). These transforms are called
+composite transforms. Nesting multiple transforms inside a single composite
+transform can make your code more modular and easier to understand.
 
-The Beam SDK comes packed with many useful composite transforms. See the API reference pages for a list of transforms:
+The Beam SDK comes packed with many useful composite transforms. See the API
+reference pages for a list of transforms:
   * [Pre-written Beam transforms for Java]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/sdk/transforms/package-summary.html)
   * [Pre-written Beam transforms for Python]({{ site.baseurl }}/documentation/sdks/pydoc/{{ site.release_latest }}/apache_beam.transforms.html)
 
-### An example of a composite transform
+#### 4.6.1. Composite transform example
```

Comment:
I think this is too robotic and the previous is too verbose; maybe

'An example [composite ]transform'